### PR TITLE
fix: do not download PRs from Kibana's official repo

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
It removes kibana entry from the list of downloadable images if the Kibana version is a PR. 

Besides that, it reworks the error handling when pulling images, as we detected that potential error was not handled, hiding pull errors. With these changes, if an error occurs when pulling an image, the error is logged with Warn level, continuing looping the array of images to pull/warm-up.

## Why is it important?
When running the tests for a kibana PR, the code tried to download all the possible images, instead of being smarter in what images to download. In this case, it was trying to download an image that does not exist: kibana/kibana:pr12345

And because kibana tag was the last one, errors in previous images caused this image not be pulled.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests (`make unit-test`), and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Author's Checklist

- [ ]

### CODE REVIEW

**Comment by mdelapenya** [comment]
Running a build for a Kibana PR here: https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-mbp/job/PR-1637/2/

**Review by kuisathaverat — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

**Review by v1v — APPROVED** [review]

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env groovy
 
-@Library('apm@current') _
+@Library('apm@test/with-docker-login') _
```
Comment (mdelapenya) [inline .ci/Jenkinsfile:3]: Will update this once the shared library gets released including https://github.com/elastic/apm-pipeline-library/pull/1306

File: .ci/Jenkinsfile [hunk 2]
```diff
@@ -348,59 +348,58 @@ def generateFunctionalTestStep(Map args = [:]){
...
-              path = "${env.WORKSPACE}\\bin;${chocoPath};${chocoPython3Path};C:\\tools\\mingw${mingwArch}\\bin;C:\\windows\\system32;C:\\windows\\system32\\WindowsPowerShell\\v1.0\\"
-              envContext.add("KIBANA_URL=https://${linuxIp}:5601")
-              envContext.add("ELASTICSEARCH_URL=https://${linuxIp}:9200")
-              envContext.add("FLEET_URL=https://${linuxIp}:8220")
-              envContext.add("PROVIDER=remote") // a remote provider is needed for Windows
-              envContext.add("OP_LOG_LEVEL=TRACE")
-              envContext.add("PATH=${path}")
-            }
-            withEnv(envContext) {
+        withDockerEnv(secret: "${DOCKER_ELASTIC_SECRET}", registry: "${DOCKER_REGISTRY}") {
```
Comment (mdelapenya) [inline .ci/Jenkinsfile:351]: `withDockerEnv` comes from https://github.com/elastic/apm-pipeline-library/pull/1306

File: e2e/_suites/fleet/ingest_manager_test.go [hunk 3]
```diff
@@ -134,14 +134,16 @@ func InitializeIngestManagerTestSuite(ctx *godog.TestSuiteContext) {
 				"docker.elastic.co/beats/elastic-agent:" + common.BeatVersion,
 				"docker.elastic.co/beats/elastic-agent-ubi8:" + common.BeatVersion,
 				"docker.elastic.co/elasticsearch/elasticsearch:" + common.StackVersion,
-				"docker.elastic.co/kibana/kibana:" + common.KibanaVersion,
 				"docker.elastic.co/observability-ci/elastic-agent:" + common.BeatVersion,
 				"docker.elastic.co/observability-ci/elastic-agent-ubi8:" + common.BeatVersion,
 				"docker.elastic.co/observability-ci/elasticsearch:" + common.StackVersion,
-				"docker.elastic.co/observability-ci/elasticsearch-ubi8:" + common.StackVersion,
 				"docker.elastic.co/observability-ci/kibana:" + common.KibanaVersion,
-				"docker.elastic.co/observability-ci/kibana-ubi8:" + common.KibanaVersion,
```
Comment (mdelapenya) [inline e2e/_suites/fleet/ingest_manager_test.go:143]: There are is no need to test Ubi8 images for the stack

File: internal/common/defaults.go [hunk 4]
```diff
@@ -141,5 +141,5 @@ func InitVersions() {
 		"BeatVersion":     BeatVersion,
 		"StackVersion":    StackVersion,
 		"KibanaVersion":   KibanaVersion,
-	}).Trace("Initial artifact versions defined")
+	}).Info("Initial artifact versions defined")
```
Comment (mdelapenya) [inline internal/common/defaults.go:144]: Always log the versions to be used

**Comment by mdelapenya** [comment]
This is ready to go now. Merging

**Pull request merged**